tools/advisory-parser/src/models.rs

```diff
@@ -0,0 +1,198 @@
+//! Advisory data models and deserialization logic.
+//! This module defines the structure of security advisory TOML files and implements
+//! custom deserialization with validation for CVE and GHSA identifiers.
+
+use lazy_static::lazy_static;
+use regex::Regex;
+use serde::de::{self, MapAccess, Visitor};
+use serde::{Deserialize, Deserializer};
+use std::fmt;
+
+lazy_static! {
```

Comment:
This is OK but the rust community has moved to [`LazyLock`](https://doc.rust-lang.org/std/sync/struct.LazyLock.html) since they added it to the std library.